pkg/filter/bloom_filter.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package filter defines and implements data structures and interfaces for skipping index.
+package filter
+
+import (
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+const (
+	k = 7
+	// B specifies the number of bits allocated for each item.
+	B = 10
```

Comment:
To identify the best composite, we should test the maximum elements of a block. The memory usage for each filter must be below 50% of the uncompressed block size. Based on this, find the lowest FPR.